blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/RunSearch.java

```diff
@@ -83,6 +84,40 @@ public Link getLink() {
         return runs;
     }
 
+    public static Iterable<BlueRun> findRuns(Job job, final Link parent, int start, int limit){
+        final List<BlueRun> runs = new ArrayList<>();
+        Iterable<Job> pipelines;
+        if(job != null){
+            pipelines = ImmutableList.of(job);
+        }else{
+            pipelines = Jenkins.getInstance().getItems(Job.class);
+        }
+
+        int skipCount=0;
+        int limitCount=0;
+        for (Job p : pipelines) {
+            RunList<? extends Run> runList = p.getBuilds();
```

Comment:
Okay ... tried it with 3000+ runs of a very simple job similar to the last time and it does seem a lot better, but it is still loading all `Run`s for the job, which I do think would end up being a problem for a more complex build with more stuff loaded up on the `Run` instance (slower to deserialize the persisted form).

I assume the improvement comes from the fact that we are now only creating the `BlueRun` instances for the number of runs needed (`limit`), which is great, but I would still consider this to be a potential issue under more real-life situations. After all, `LazyBuildMixIn` etc would appear to have been created exactly for the purpose of avoiding this "loading everything" situation.